gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/RetryingTest.java

```diff
@@ -396,4 +448,35 @@ public static UnaryCallSettings<Integer, Integer> createSettings(
         .setRetrySettings(retrySettings)
         .build();
   }
+
+  private ApiTracerFactory getApiTracerFactory() {
+    ApiTracer tracer =
+        new BaseApiTracer() {
+          @Override
+          public void attemptFailed(Throwable error, Duration delay) {
+            tracerAttemptsFailed.incrementAndGet();
+            super.attemptFailed(error, delay);
+          }
+
+          @Override
+          public void attemptStarted(int attemptNumber) {
+            tracerAttempts.incrementAndGet();
+            super.attemptStarted(attemptNumber);
+          }
+
+          @Override
+          public void operationFailed(Throwable error) {
+            tracerOperationFailed.set(true);
+            super.operationFailed(error);
+          }
+
+          @Override
+          public void attemptFailedRetriesExhausted(Throwable error) {
+            tracerFailedRetriesExhausted.set(true);
+            super.attemptFailedRetriesExhausted(error);
+          }
+        };
+    ApiTracerFactory tracerFactory = (parent, spanName, operationType) -> tracer;
+    return tracerFactory;
+  }
```

Comment:
I wonder if this could be refactored into something like a TestTracerFactory + TestApiTracer that basically does this. Increments the value and exposes getters for the values.

Not something that needs to be done in this PR, but we can keep in mind in case we duplicate this in other unit tests

Reply:
May take a stab at it when implementing our yet-to-be made decision on the failing test https://github.com/googleapis/sdk-platform-java/pull/3300#discussion_r1815749326

Reply:
Done, now we have a separate test api tracer + factory.